libraries/timeslider/src/components/TimeSlider/TimeSliderModel.ts

```diff
@@ -0,0 +1,94 @@
+import {
+    ComponentModelBase,
+    ComponentModelProperties,
+    importModel,
+    PropertyDefs,
+    serializable,
+} from "@vertigis/web/models";
+import { ItemType } from "@vertigis/arcgis-extensions/ItemType";
+import { TimeIntervalUnit, TimeSliderLayout, TimeSliderMode } from "./designer";
+import { MapModel } from "@vertigis/web/mapping/MapModel";
+import EsriTimeSlider from "@arcgis/core/widgets/TimeSlider";
+
+interface TimeSliderModelProperties extends ComponentModelProperties {
+    layout?: TimeSliderLayout;
+    loop?: boolean;
+    playRate?: number;
+    mode?: TimeSliderMode;
+    overrideStops?: boolean;
+    timeInterval?: number;
+    timeIntervalUnit?: TimeIntervalUnit;
+    timeVisible?: boolean;
+    inheritedWidgetConfig?: boolean;
```

Comment:
All properties except `inheritedWidgetConfig` are configurable by the end user. What this does is it makes sure that the time slider config defaults (for the loop and playRate properties) are only used the first time the widget is loaded. After that, the time slider component's model will always be applied to the time slider widget. The reason why this is serialized is I need this flag to persist between app designer sessions so the component remembers whether it has inherit config before or not.

Reply:
I see. I think that this is a bit of an unusual pattern compared to everything else in Web. I would like to try and find a solution that does not involve serializing out this flag to app.json, since it was not really meant for that purpose.

The other thing that occurred to me is that web maps have their own config for many of these values (see [here](https://developers.arcgis.com/web-map-specification/objects/timeSlider/)). Mathew has a PR up to consume time extent from the web map. It might be worth to have the three of us meet and chat about how to coordinate all of this. Normally when there is a way to configure something in Esri's config, we do not want to duplicate it in our own config.

Reply:
Right, I agree it's a weird pattern though it was the only way I could think of "remembering" whether the time slider config has been inherited from the config that Mathew's PR sets up. I think a meeting would be a good idea.

Reply:
As @eric-parsons pointed out above, I had imagined that we wouldn't want to (de)serialize any properties that exist on the Webmap object - I think these properties should always be read off the Webmap (or fall back on data-driven defaults at runtime, if not explicitly configured). Any additional properties that exist only on the ESRI Widget would be good candidates for this model's serialization.

If the user were to programmatically configure any of the properties that are shared between Widget and Webmap at runtime, I would expect those changes not to be saved to the app state.